packages/ckeditor5-find-and-replace/src/ui/findandreplaceformview.js

```diff
@@ -47,190 +47,133 @@ export default class FindAndReplaceFormView extends View {
 		 *
+		 * @private
 		 * @readonly
-		 * @member {module:utils/focustracker~FocusTracker}
+		 * @observable
+		 * @member {Boolean} #isCounterHidden
 		 */
-		this.focusTracker = new FocusTracker();
+		this.set( 'isCounterHidden', true );
 
 		/**
-		 * An instance of the {@link module:utils/keystrokehandler~KeystrokeHandler}.
+		 * Assigning form elements views.
 		 *
+		 * @private
 		 * @readonly
-		 * @member {module:utils/keystrokehandler~KeystrokeHandler}
+		 * @observable
+	 	 * @param {Function} t Used to translate message to the uiLanguage.
 		 */
-		this.keystrokes = new KeystrokeHandler();
+		this._assignViews( t );
 
 		/**
-		 * A collection of views that can be focused in the form.
+	 	 * Initialization of focusables elements of the form.
 		 *
+		 * @private
 		 * @readonly
-		 * @protected
-		 * @member {module:ui/viewcollection~ViewCollection}
-		 */
-		this._focusables = new ViewCollection();
-
-		/**
-		  * Helps cycling over {@link #_focusables} in the form.
-		  *
-		  * @readonly
-		  * @protected
-		  * @member {module:ui/focuscycler~FocusCycler}
-		  */
-		this._focusCycler = new FocusCycler( {
-			focusables: this._focusables,
-			focusTracker: this.focusTracker,
-			keystrokeHandler: this.keystrokes,
-			actions: {
-				// Navigate form fields backwards using the <kbd>Shift</kbd> + <kbd>Tab</kbd> keystroke.
-				focusPrevious: 'shift + tab',
-
-				// Navigate form fields forwards using the <kbd>Tab</kbd> key.
-				focusNext: 'tab'
+		 * @observable
+	 	 */
+		this._initFocusables();
+
+		this.bind( 'isDirty' ).to(
+			this, 'searchText', state, 'searchText',
+			this.matchCaseView, 'isChecked', state, 'matchCase',
+			this.matchWholeWordsView, 'isChecked', state, 'matchWholeWords',
+			( formSearchText, stateSearchText, formMatchCase, stateMatchCase, formMatchWholeWords, stateMatchWholeWords ) => {
+				return formSearchText !== stateSearchText ||
+				formMatchCase !== stateMatchCase ||
+				formMatchWholeWords !== stateMatchWholeWords;
 			}
-		} );
+		);
 
-		this.bind( 'searchText' ).to( this.findInputView.fieldView, 'value' );
+		this.bind( 'searchText' ).to( this.findInputView.fieldView, 'value', enforceStringValue );
 		this.findButtonView.bind( 'isEnabled' ).to( this.findInputView.fieldView, 'isEmpty', value => !value );
-		this.bind( 'replaceText' ).to( this.replaceInputView.fieldView, 'value' );
+		this.bind( 'replaceText' ).to( this.replaceInputView.fieldView, 'value', enforceStringValue );
 		this.replaceButtonView.bind( 'isEnabled' ).to( this, 'isSearching' );
 		this.replaceAllButtonView.bind( 'isEnabled' ).to( this, 'isSearching' );
 
 		this.bind( 'isCounterHidden' ).to( this, 'matchCount', this, 'highlightOffset', ( matchCount, highlightOffset ) => {
-			return matchCount === null || matchCount === 0 ||
-				highlightOffset === null || highlightOffset === 0;
+			return matchCount === null || highlightOffset === null;
 		} );
 
+		this.bind( 'isSearching' ).to( this );
+
+		if ( state ) {
```

Comment:
State is mandatory, there can be no situation when state is not passed.

So this if is not needed.

Reply:
Also this means that you won't have to do two `isSearching` bindings and later unbinding. Instead you'll bind `isSearching` just once.